dimod/serialization/bson.py

```diff
@@ -52,19 +64,26 @@ def mul(a, b):
     else:
         vals = _vals
 
+    lvals, qvals = lin.tobytes(), vals.tobytes()
+    if PY2:
+        lvals, qvals = bson.Binary(lvals), bson.Binary(qvals)
+
     doc = {
         "as_complete": as_complete,
-        "linear": lin.tobytes(),
-        "quadratic_vals": vals.tobytes(),
+        "linear": lvals,
+        "quadratic_vals": qvals,
         "variable_type": "SPIN" if bqm.vartype == bqm.SPIN else "BINARY",
         "offset": off,
         "variable_order": variable_order,
         "index_dtype": np.dtype(index_dtype).str,
     }
 
     if not as_complete:
-        doc["quadratic_head"] = i.tobytes()
-        doc["quadratic_tail"] = j.tobytes()
+        ii, jj = i.tobytes(), j.tobytes()
+        if PY2:
+            ii, jj = bson.Binary(ii), bson.Binary(jj)
```

Comment:
It might be better to make a function `tobytes` in `dimod/compatibility23.py` or something like that, just to make it a little cleaner.